pkg/cmd/cli/secrets/basicauth.go

```diff
@@ -0,0 +1,219 @@
+	cmd.Flags().StringVar(&o.CertificatePath, "ca-cert", "", "Path to the certificate file")
+	cmd.Flags().StringVar(&o.GitConfigPath, "gitconfig", "", "Path to the .gitconfig file")
+	cmd.Flags().BoolVarP(&o.PromptForPassword, "prompt", "", false, "Prompt for password or token")
+
+	// autocompletion hints
+	cmd.MarkFlagFilename("ca-cert")
+	cmd.MarkFlagFilename("gitconfig")
+
+	kcmdutil.AddPrinterFlags(cmd)
+
+	return cmd
+}
+
+// CreateBasicAuthSecret saves created Secret structure and prints the secret name to the output on success.
+func (o *CreateBasicAuthSecretOptions) CreateBasicAuthSecret() error {
+	secret, err := o.MakeBasicAuthSecret()
+	if err != nil {
+		return err
+	}
+
+	if _, err := o.SecretsInterface.Create(secret); err != nil {
+		return err
+	}
+
+	fmt.Fprintf(o.GetOut(), "secret/%s\n", secret.Name)
+	return nil
+}
+
+// MakeBasicAuthSecret builds up the Secret structure containing secret namespace, name, type and data structure
+// containing desired credentials.
+func (o *CreateBasicAuthSecretOptions) MakeBasicAuthSecret() (*api.Secret, error) {
+	secret := &api.Secret{}
+	secret.Namespace = o.SecretNamespace
+	secret.Name = o.SecretName
+	secret.Type = api.SecretTypeOpaque
+	secret.Data = map[string][]byte{}
+
+	if len(o.Username) != 0 {
+		secret.Data[SourceUsername] = []byte(o.Username)
+	}
+
+	if len(o.Password) != 0 {
+		secret.Data[SourcePassword] = []byte(o.Password)
+	}
+
+	if len(o.CertificatePath) != 0 {
+		caContent, err := ioutil.ReadFile(o.CertificatePath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data[SourceCertificate] = caContent
+	}
+
+	if len(o.GitConfigPath) != 0 {
+		gitConfig, err := ioutil.ReadFile(o.GitConfigPath)
+		if err != nil {
+			return nil, err
+		}
+		secret.Data[SourceGitConfig] = gitConfig
+	}
+
+	return secret, nil
+}
+
+// Complete fills CreateBasicAuthSecretOptions fields with data and checks for mutual exclusivity
+// between flags from different option groups.
+func (o *CreateBasicAuthSecretOptions) Complete(f *kcmdutil.Factory, args []string) error {
+	if len(args) != 1 {
+		return errors.New("must have exactly one argument: secret name")
+	}
+	o.SecretName = args[0]
+
+	if len(o.Username) == 0 && len(o.Password) == 0 && len(o.CertificatePath) == 0 {
+		return errors.New("must provide basic authentication credentials")
+	}
+
+	if o.PromptForPassword {
+		if len(o.Password) != 0 {
+			return errors.New("must provide either --prompt or --password flag")
+		}
+		if cmdutil.IsTerminal(o.Reader) {
+			o.Password = cmdutil.PromptForPasswordString(o.Reader, o.Out, "Password: ")
+			if len(o.Password) == 0 {
+				return errors.New("password must be provided")
+			}
+		} else {
+			return errors.New("provided reader is not a terminal")
+		}
+	}
+
+	client, err := f.Client()
+	if err != nil {
+		return err
+	}
+	o.SecretNamespace, _, err = f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	o.SecretsInterface = client.Secrets(o.SecretNamespace)
```

Comment:
I think we should just use `f.DefaultNamespace()` when creating `o.SecretsInterface` and leave the namespace empty in the generated secret itself so it can be piped to `oc create -f ...` in any namespace